File: solutions/cpp/0059.cpp
class Solution {
public:
    vector<vector<int>> generateMatrix(int n) {
        vector<vector<int>> ret(n, vector<int>(n));
        int count = 1;

        for (int min = 0, max = n - min - 1; min < n / 2; min++, max--) {
            for (int i = min; i < max; i++) ret[min][i] = count++;
            for (int i = min; i < max; i++) ret[i][max] = count++;
            for (int i = max; i > min; i--) ret[max][i] = count++;
            for (int i = max; i > min; i--) ret[i][min] = count++;
        }

        if (n & 1) ret[n / 2][n / 2] = count;

        return ret;
    }
};

File: solutions/python3/0059.py
class Solution:
    def generateMatrix(self, n: int) -> List[List[int]]:
        ret = [[0 for j in range(n)] for i in range(n)]
        count = 1

        for min in range(n // 2):
            max = n - min - 1
            for i in range(min, max):
                ret[min][i] = count
                count += 1
            for i in range(min, max):
                ret[i][max] = count
                count += 1
            for i in range(max, min, -1):
                ret[max][i] = count
                count += 1
            for i in range(max, min, -1):
                ret[i][min] = count
                count += 1

        if n & 1:
            ret[n // 2][n // 2] = count

        return ret